core/rawdb/rawdb.go

```diff
@@ -0,0 +1,63 @@
+package rawdb
+
+import (
+	"encoding/json"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/types"
+	"github.com/ChainSafe/gossamer/polkadb"
+	log "github.com/ChainSafe/log15"
+)
+
+// check checks to see if there an error if so writes err + message to terminal
+func check(e error, msg string) {
+	if e != nil {
+		log.Crit(msg, "err", e)
+	}
+}
+
+// SetHeader stores a block header into the KV-store; key is headerPrefix + hash
+func SetHeader(db polkadb.Writer, header *types.BlockHeader) {
+	hash := header.Hash
+
+	// Write the encoded header
+	bh, err := json.Marshal(header)
+	check(err, "Failed to encode header to bytes")
+
+	err = db.Put(headerKey(hash), bh)
+	check(err, "Failed to store header")
+}
+
+// GetHeader retrieves block header from KV-store using headerKey
+func GetHeader(db polkadb.Reader, hash common.Hash) types.BlockHeader {
```

Comment:
Codeclimate might have a point here. You could write a function `get(db polkadb.Reader, hash common.Hash, out interface{})` where `out` is `result` as defined here, and then use that everywhere.